src/containerapp/azext_containerapp/_cloud_build_utils.py

```diff
@@ -177,12 +174,35 @@ def spin():
         done_spinner = False
         thread = display_spinner("Streaming Cloud Build logs")
         headers = {'Authorization': 'Bearer ' + token}
-        response_log_streaming = requests.get(
-            log_streaming_endpoint,
-            headers=headers,
-            stream=True)
-        if not response_log_streaming.ok:
-            raise ValidationError(f"Error when streaming the logs, request exited with {response_log_streaming.status_code}")
+        logs_stream_retries = 0
+        maximum_logs_stream_retries = 5
+        while logs_stream_retries < maximum_logs_stream_retries:
+            logs_stream_retries += 1
+            response_log_streaming = requests.get(
+                log_streaming_endpoint,
+                headers=headers,
+                stream=True)
+            if not response_log_streaming.ok:
```

Comment:
Yes, so basically there are two different steps:
1. First, we call the log streaming endpoint to get the logs. This should always work, so if it fails it's OK to raise an exception immediately and stop the flow as it's a real issue we need to investigate.
2. Second, even when the log streaming call succeeded, we need to make sure that the logs are actually flowing. This is something that we know fails sometimes (with the "Kubernetes error happened" error), and this is where the retry comes into action. We have plans to improve the resiliency at the service level so that this error never happens (and so we wouldn't even need the retry), but as of today we still need it for this second part.